src/ServerExtension.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Panther project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Panther;
+
+use PHPUnit\Runner\AfterLastTestHook;
+use PHPUnit\Runner\BeforeFirstTestHook;
+
+final class ServerExtension implements BeforeFirstTestHook, AfterLastTestHook
+{
+    public function executeBeforeFirstTest(): void
+    {
+        echo "Starting Panther server...\n";
+        PantherTestCaseTrait::$stopServerOnTeardown = true;
+        PantherTestCaseTrait::startWebServer();
```

Comment:
Instead of duplicating code, can we use a trait for these 3 lines and the ones from the other method, and then use the trait in both the `ServerExtension` class and `ServerListener`?

Reply:
Done.